locals.tf

```diff
@@ -6,6 +6,7 @@ locals {
 
   # If var.role_name contains one instance of "%s", use format()
   # to replace the "%s" with var.hostname, otherwise just use
-  # var.role_name as is
-  role_name = length(regexall(".*%s.*", var.role_name)) > 0 ? format(var.role_name, var.hostname) : var.role_name
+  # var.role_name as is.  If the role name is longer than 64 characters
+  # (the current AWS limit), truncate it to the first 64 characters.
+  role_name = substr(length(regexall(".*%s.*", var.role_name)) > 0 ? format(var.role_name, var.hostname) : var.role_name, 0, 63)
```

Comment:
Shouldn't this be `substr(..., 0, 64)`?  The final parameter is [the length of the substring, not the final index](https://www.terraform.io/language/functions/substr).

Reply:
Aww snap, you are right- I totally misread that, great catch!  Fixed in https://github.com/cisagov/cert-read-role-tf-module/pull/43/commits/82a1f24d3cba018e155e6d6f1b19cf83b79200e9.